metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1481,11 +1754,24 @@ def emit_workbook_as_container(self, workbook: Dict) -> Iterable[MetadataWorkUni
             if (tag_list and self.config.ingest_tags)
             else None
         )
+        parent_key = None
```

Comment:
this method is already FAR too long. can we isolate this code and add comments to explain what it's doing?

Reply:
Can we take it later as part of code refactoring ?

Reply:
let's make sure we get to this refactoring later on